### DISCUSSION TITLE
Delete all recordings from one specific species

### DISCUSSION THREAD

**Discussion by Fish31520** [mcguirepr89/BirdNET-Pi#612] [Q&A] [answered]
Hi,

I often got wrong identifications when people are speaking in the street, just below my mic (they are usually labeled as Botaurus stellaris 😄).

I could go in the tools and exclude some species but I'm wondering if there's also a way to delete all the recording of a given species at once.

Thanks for your help !

**Comment by ehpersonal38** [comment, marked as answer]
Hi @Fish31520 

There is certainly a way of doing this if you're willing to get a little hacky. The basic procedure is executing the following SQL command in your database: `DELETE FROM detections WHERE Com_Name = "Northern Cardinal";`

I've created the following shortcut that should take you to the right place in your local installation's Database Maintenance tool to execute the above command. You'll need to login with your BirdNET-Pi credentials and click "Login" once. Hoping it works for you.
http://birdnetpi.local/scripts/adminer.php?sqlite=&username=&db=birds.db&sql=DELETE%20FROM%20detections%20where%20Com_Name%20%3D%20%22TEST%22%3B

Replace "TEST" with the common name of the species you wish to delete, and click "Execute" at the bottom to run the command. This will completely purge that species from your database. The audio files will be automatically cleaned up by the disk_check cronjob probably within a week or so.

**Reply by Fish31520** [reply]
Hi @ehpersonal38 and thanks for that wonderful answer, it's working pretty well.
I did a list of all the species misidentified and with your tool the job is done really fast !
Thanks !

**Comment by mcguirepr89** [comment]
Hello -- sorry for the late addition, here, but I did want to let you guys know about a little tool I put as shell `alias` called "`remove`" which does the file-cleanup part of this (@ehpersonal38 was spot on with cleaning up the database)

`remove` does the following, which I would like to just briefly outline, (so be careful!):
```
function remove() {
  source /etc/birdnet/birdnet.conf
  find $EXTRACTED -iname *${1}* -exec rm -drfv {} +
}
```

Thus, if you were to type: `remove 'cardinal'`, the `remove` function would look through ALL of your extracted recordings for ANY file that contained (_case INsensitive_) 'cardinal'.

That is harmless enough. Though if you aren't explicit enough, you could potentially do some damage. For instance, if I wanted to get rid of all of the "House Finch" recordings, I might think that I could accomplish that a little more quickly with `remove 'house'`, though that would actually remove ANY species that had "House" in the title (House Sparrow, House Finch, House Wren, House Swift, etc.).

Anywho, perhaps that tools is helpful, and perhaps it is just easier to leave it all alone! Your call!

My best,
Patrick

**Reply by Fish31520** [reply]
Thanks @mcguirepr89,


If I get it right, in that case the web terminal need to be used ?
